utils/proc/process.go

```diff
@@ -0,0 +1,208 @@
+// This of course will be a point-in-time snapshot of when this method was
+// called. Some operating systems don't provide snapshot capability of the
+// process table, in which case the process table returned might contain
+// ephemeral entities that happened to be running when this was called.
+func Ps(ctx context.Context) (processes []IProcess, err error) {
+	pss, err := process.ProcessesWithContext(ctx)
+	err = ConvertProcessError(err)
+	if err != nil {
+		return
+	}
+	for i := range pss {
+		processes = append(processes, wrapProcess(pss[i]))
+	}
+	return
+}
+
+// FindProcess looks up a single process by pid.
+//
+// Process will be nil and error will be commonerrors.ErrNotFound if a matching process is
+// not found.
+func FindProcess(ctx context.Context, pid int) (p IProcess, err error) {
+	p, err = NewProcess(ctx, pid)
+	if commonerrors.Any(err, nil, commonerrors.ErrTimeout, commonerrors.ErrCancelled) {
+		return
+	}
+	err = fmt.Errorf("%w: process (#%v) could not be found: %v", commonerrors.ErrNotFound, pid, err.Error())
+	return
+}
+
+type ps struct {
+	imp *process.Process
+}
+
+func (p *ps) Cmdline() string {
+	cmd, _ := p.imp.Cmdline()
+	return cmd
+}
+
+func (p *ps) Cwd() string {
+	cwd, _ := p.imp.Cwd()
+	return cwd
+}
+
+func (p *ps) Parent() IProcess {
+	pp, _ := p.imp.Parent()
+	return wrapProcess(pp)
+}
+
+func (p *ps) Children(ctx context.Context) (children []IProcess, err error) {
+	err = parallelisation.DetermineContextError(ctx)
+	if err != nil {
+		return
+	}
+	cp, err := p.imp.Children()
+	err = ConvertProcessError(err)
+	if err != nil {
+		return
+	}
+	for i := range cp {
+		children = append(children, wrapProcess(cp[i]))
+	}
+	return
+}
+
+func (p *ps) Name() string {
+	n, _ := p.imp.Name()
+	return n
+}
+
+func (p *ps) Environ(ctx context.Context) (env []string, err error) {
+	env, err = p.imp.EnvironWithContext(ctx)
+	err = ConvertProcessError(err)
+	return
+}
+
+func (p *ps) Pid() int {
+	return int(p.imp.Pid)
+}
+
+func (p *ps) PPid() int {
+	ppid, _ := p.imp.Ppid()
+	return int(ppid)
+}
+
+func (p *ps) Executable() string {
+	ex, _ := p.imp.Exe()
+	return ex
+}
+
+func (p *ps) Terminate(ctx context.Context) error {
+	err := ConvertProcessError(p.imp.TerminateWithContext(ctx))
+	if commonerrors.Any(err, commonerrors.ErrNotFound) {
+		err = nil
+	}
+	return err
+}
+
+func (p *ps) KillWithChildren(ctx context.Context) error {
+	return killProcessAndChildren(ctx, p.imp)
+}
+
+func killProcessAndChildren(ctx context.Context, p *process.Process) (err error) {
+	err = parallelisation.DetermineContextError(ctx)
+	if err != nil {
+		return
+	}
+	if p == nil {
+		return
+	}
+	defer func() {
+		_ = p.Kill()
+	}()
+	err = ConvertProcessError(p.TerminateWithContext(ctx))
+	if err != nil {
+		if commonerrors.Any(err, commonerrors.ErrNotFound) {
+			err = nil
+		}
+		return
+	}
+	err = ConvertProcessError(killGroup(ctx, p.Pid))
+	if err == nil {
+		if commonerrors.Any(err, commonerrors.ErrNotFound) {
+			err = nil
+		}
+		return
```

Comment:
I think you can use our helpers to simplify this.

```suggestion
		err = commonerrors.Ignore(err, commonerrors.ErrNotFound)
		return
```

This also applied to the other places where this is used